### PR TITLE
Add protecting api blog

### PR DESCRIPTION
**Pull request by mooreds**
Questions:

* is this better to split up into two blog posts (one about centralized/decentralized, one about first/third party)
* is the term 'central authority' useful, or should I change it
* any transitions seem abrupt or don't make sense
* is it obvious how to change the CSS for the parallel sequence diagrams? I couldn't figure it out

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206332808576804
  - https://app.asana.com/0/0/1206341940874047

### CODE REVIEW

File: astro/src/content/blog/protecting-your-api.mdx [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+But what if Pied Piper builds a platform to foster an ecosystem of todos based add-ons. If this takes off, there are additional services which will make users' task management experience more enjoyable. Some examples of third-party todo based enhancements include:
+
+* an LLM based service to summarize todos
+* a reminder service to read todo due dates and text or email a user about upcoming tasks
+* an analytics service to give a user monthly summaries of their accomplishments
+* custom clients to add todos using an Apple watch interface
+
+While all these services could be built by Pied Piper, if they open up a platform and provide the right incentives, they can harness the creativity and energy of more developers than they could ever hope to hire.
+
+With such a platform in mind, Pied Piper needs to authenticate not just the API requester, but also control access in a more sophisticated way, based on what each service needs. For example, the Apple watch client needs to be able to read and write todos, but the analytics service only needs to be able to read them.
```
Comment (alex-fusionauth) [inline astro/src/content/blog/protecting-your-api.mdx:241]:
```suggestion
With such a platform in mind, Pied Piper needs to authenticate not just the API requester, but also control access in a more sophisticated way, based on what each service needs. For example, the Apple Watch client needs to be able to read and write todos, but the analytics service only needs to be able to read them.
```

File: astro/src/content/blog/protecting-your-api.mdx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+The central authority should also verify requested scopes. Since the scope request is under control of the third-party client, a malicious or buggy client could request disallowed values. If the central authority receives an initial scope request that was not granted to the third-party, the request is invalid and should not be processed.
+
+Finally, if you are using scopes in a decentralized API key verification system, they should be validated by the API server to check that the scope in the API token matches the requested operation.
+
+### Designing Scopes
+
+Scope design is hard when first designing APIs, but even more painful to retrofit. Since third-party clients are building on top of the APIs you provide, updating or changing scopes will be a long, drawn out process.
+
+Spend some time up front thinking about what scopes make sense for your API. You want them to:
```
Comment (alex-fusionauth) [inline astro/src/content/blog/protecting-your-api.mdx:289]:
```suggestion
Spend some time upfront thinking about what scopes make sense for your API. You want them to:
```

File: astro/src/content/blog/protecting-your-api.mdx [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+* `todo-read`: the permission to read and list todos
+* `todo-write`: the permission to create and update todos
+
+As more API functionality is built, different read and write scopes can be added. It's also easier to add new scopes than to remove them, so err on the side of minimalism. This is more of an art than a science.
+
+## Wrapping Up
+
+When you are exposing an API, you need to protect it.
+
+You should think about whether to use a centralized API key verification system or decentralized verification. The centralized option requires constant communication between the consumers of API keys, the systems serving the API, and the central authority issuing API keys, all to ensure validity of the keys.
```
Comment (alex-fusionauth) [inline astro/src/content/blog/protecting-your-api.mdx:310]:
```suggestion
You should think about whether to use a centralized API key verification system or decentralized verification. The centralized option requires constant communication between the consumers of API keys, the systems serving the API, and the central authority issuing API keys, all to ensure the validity of the keys. 
```

**Review by alex-fusionauth — CHANGES_REQUESTED** [review]
- is this better to split up into two blog posts (one about centralized/decentralized, one about first/third party)
-- No I like this all in one
- is the term 'central authority' useful, or should I change it
-- yes it is useful and very understandable in the charts
- any transitions seem abrupt or don't make sense
- is it obvious how to change the CSS for the parallel sequence diagrams? I couldn't figure it out
-- I don't like the red at all, I am also unclear on what the `par` keyword is and its hidden on the diagram

Added a CSS question for Andy
https://inversoft.slack.com/archives/C02CR2M6SRK/p1706729482035529

**Review by bradmccarty — COMMENTED** [review]
This is good to go from an SEO perspective @mooreds

File: astro/src/content/blog/protecting-your-api.mdx [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+
+## Authentication And Authorization
+
+Typically after you authenticate a request and know who the caller is, you authorize it. This lets the API server know what access the caller has or if access should be denied.
+
+<Aside type="note" nodark="true">
+Authentication is "who are you" and authorization is "what can you do". While they can be treated separately (car keys authorize without authenticating and a business card authenticates without authorizing), they are often considered together.
+</Aside>
+
+Authorization can be done in a number of ways, including role-based access control (RBAC), attribute-based access control (ABAC), and relationship-based access control (ReBAC). Authorization decisions are typically embedded in business logic or extracted to a standalone policy decision point (PDP).
```
[the remark this answers is not in the document]
Comment (mooreds) [reply astro/src/content/blog/protecting-your-api.mdx:106]: Agreed, but our primary solution for ABAC and ReBAC is "use an authorization server" like Cerbos or Permit.

**Review by alex-fusionauth — APPROVED** [review]
I would still rock a different image, but the rest looks good. Nice work on the CSS update.

**Pull request merged**